gtsam/hybrid/HybridFactorGraph.cpp

```diff
@@ -0,0 +1,58 @@
+//
+// Created by Fan Jiang on 3/11/22.
+//
+
+#include <gtsam/hybrid/HybridEliminationTree.h>
+#include <gtsam/hybrid/HybridJunctionTree.h>
+#include <gtsam/hybrid/HybridFactorGraph.h>
+
+#include <gtsam/hybrid/HybridGaussianFactor.h>
+
+#include <gtsam/inference/EliminateableFactorGraph-inst.h>
+
+namespace gtsam {
+
+template class EliminateableFactorGraph<HybridFactorGraph>;
+
+/* ************************************************************************ */
+std::pair<HybridConditional::shared_ptr, HybridFactor::shared_ptr>  //
+EliminateHybrid(const HybridFactorGraph& factors,
```

Comment:
Because we are in the Conditional Gaussian regime there are only
few cases:
* continuous variable, we make a GM if there are hybrid factors;
* continuous variable, we make a GF if there are no hybrid factors;
* discrete variable, no continuous factor is allowed (escapes CG regime), so we panic
* if discrete only we do the discrete elimination.

We can know which variable is discrete by querying the factors.